### ISSUE TITLE
Get aggregate state

### ISSUE THREAD

**Issue by slashdotdash** [commanded/commanded#336]
An aggregate's current state can be accessed using the [`Commanded.Aggregates.Aggregate.aggregate_state/3` function](https://github.com/commanded/commanded/blob/3b8f08526b97a196e3f94e7c868cc059c9062652/lib/commanded/aggregates/aggregate.ex#L140). You pass the Commanded application, aggregate module, and aggregate identity:

```elixir
alias Commanded.Aggregates.Aggregate

%BankAccount{..} = Aggregate.aggregate_state(BankApp, BankAccount, account_number)
```

However this only works when the aggregate process has already been started by having a command dispatched to it. 

A helper function should be added to a Commanded application to allow an aggregate's state to be queried.

```elixir
{:ok, %BankAccount{}} = BankApp.aggregate_state(BankAccount, account_number)
```

Or using `Commanded.aggregate_state/3` where the application is provided as an argument:

```elixir
{:ok, %BankAccount{}} = Commanded.aggregate_state(BankApp, BankAccount, account_number)
```

Retrieving aggregate state should be done by querying the event store for an optional state snapshot and then replaying the aggregate's event stream. It is not necessary to start an aggregate `GenServer` process.

**Comment by jwilger** [comment]
> A helper function should be added to a Commanded application to allow an aggregate's state to be queried. 

I'm curious about the motivation for that. Isn't directly querying an aggregate's state exactly what one is *not* supposed to do when using the CQRS pattern?

**Comment by slashdotdash** [comment]
@jwilger Yes, it breaks the strict CQRS interpretation of defining a separate model for writes and reads. I would recommend having this separation for most use cases.

This is mostly designed to be a convenience helper function for those scenarios where read access patterns can be easily fulfilled by querying the aggregate state by its identity. An example from a recent project is where we use an isolated tenant-specific event store for each customer's data and didn't want or need to build a separate read model for certain aggregates. Accessing the aggregate state directly was good enough.

**Comment by slashdotdash** [comment]
Closed by #448.
